### PR TITLE
front: refacto simulation ts rtk

### LINKED ISSUE #4715: front: migrate RTK endpoint in simulation.js and migrate to TS
### Description and goal

Use RTK for endpoints calls

### Acceptance criteria

The endpoints are all called with RTK
File written in ts

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #4715 
* simulation: 
  * remove progressiveDuplicateTrain (unused)
  * updateSimulation (duplicate in actions.ts)

### CODE REVIEW

File: front/src/reducers/index.ts
Comment (clarani) [inline front/src/reducers/index.ts:1]: :clap:

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx [hunk 1]
```diff
@@ -149,7 +151,7 @@ const Map: FC<MapProps> = ({ setExtViewport }) => {
         const trainTime = trainRegime.head_positions[0][0].time;
         const train2ndTime = last(last(trainRegime.head_positions))?.time as number;
         if (
-          actualTime >= trainTime &&
+          actualTime >= (trainTime as number) &&
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: Are you sure you need this cast ? I removed it and no warning appeared
```suggestion
          actualTime >= trainTime &&
```

File: front/src/reducers/osrdsimulation/simulation.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    const apiDetailsForNextTrain = nextTrain
+      ? JSON.stringify(getTrainDetailsForAPI(nextTrain))
+      : undefined;
+
+    // This trains is absent from the future simulation state.
+    if (!nextTrain) {
+      // Call delete API (await)
+      try {
+        // TODO: REMOVE THIS AFTER REFACTORING
+        //   deleteRequest(`${trainscheduleURI}${id}/`);
```
Comment (clarani) [inline front/src/reducers/osrdsimulation/simulation.ts]:
```suggestion
```

**Review by clarani — APPROVED** [review]
Thanks a lot for this PR, you did a great job migrating it ! I hope this will allow to avoid future bugs.

Here is a ticket to refacto the simulation file later : https://github.com/osrd-project/osrd/issues/5049

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx (hunk 1 above)
Comment (Uriel-Sautron) [reply front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: Done

File: front/src/reducers/osrdsimulation/simulation.ts (hunk 2 above)
Comment (Uriel-Sautron) [reply front/src/reducers/osrdsimulation/simulation.ts]: Done

**Pull request merged**